### PR TITLE
Feature: Rename Jails

### LINKED ISSUE #52: Feature: Rename jail
A jails name also defines the jails dataset location relative to the activated zpool. That's why it's important to do more than just renaming the property. Certainly we must force a jail to be stopped for renaming it.

### Requirements
- When renaming a Jail
  - [ ] an exception is raised when the jail is started
  - [ ] moves the dataset immediately
  - [ ] changes the `host_hostname` when it was manually set (available in config.data) and matches the old jail name

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/48866814-feature-rename-jail?utm_campaign=plugin&utm_content=tracker%2F70027116&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F70027116&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gronke**
closes #52

### Changes
- Introduced ioc command `rename`
- Renames a dataset if the jail was stopped and the new name is valid

### Usage
```
ioc rename my-jail new-jail-name
```

### CODE REVIEW

File: libiocage/lib/Jail.py [hunk 1]
```diff
@@ -289,6 +289,23 @@ def destroy(self, force=False):
 
         self.storage.delete_dataset_recursive(self.dataset)
 
+    def rename(self, new_name: str):
```
Comment (igalic) [inline libiocage/lib/Jail.py]: please add `-> None` as return type

File: libiocage/lib/Jail.py [hunk 2]
```diff
@@ -289,6 +289,23 @@ def destroy(self, force=False):
...
+        """
+        Change the name of a jail
+        """
+
+        self.require_jail_existing()
+        self.require_jail_stopped()
+
+        current_id = self.config["id"]
+        dataset = self.dataset
+        self.config["name"] = new_name  # validates new_name
```
Comment (igalic) [inline libiocage/lib/Jail.py]: huh, how??

File: libiocage/cli/rename.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def cli(ctx, jail, name):
+    """
+    Rename a stopped jail.
+    """
+
+    logger = ctx.parent.logger
+    try :
+        ioc_jail = libiocage.lib.Jail.Jail(jail, logger=logger)
+        ioc_jail.rename(name)
+    except libiocage.lib.errors.IocageException:
```
Comment (igalic) [inline libiocage/cli/rename.py]: what kind of output will this produce?

**Comment by igalic** [comment]
unrelated to this, travis is unhappy:

```
+./libiocage/lib/JailConfigResolver.py:86:18: F821 undefined name 'isintance'
+./libiocage/lib/JailConfig.py:617:13: F841 local variable 'no_user_defined_property' is assigned to but never used
```

we should fix this first before merging anything else

File: libiocage/lib/Jail.py (hunk 2 above)
Comment (skarekrow) [reply libiocage/lib/Jail.py]: It should throw a KeyError I believe, but if so it's missing a try block.

**Pull request merged**